checkstyle.xml

```diff
@@ -0,0 +1,85 @@
+<?xml version="1.0"?>
+
+<!DOCTYPE module PUBLIC
+      "-//Puppy Crawl//DTD Check Configuration 1.1//EN"
+      "http://www.puppycrawl.com/dtds/configuration_1_1.dtd">
+
+<!-- Created from https://wiki.jasig.org/display/UPM40/Code+Style+and+Conventions -->
+<module name="Checker">
+  <!--
+    The default severity to rules is "error".
+    If false positives are being flagged, individual module severity can be
+    loosened by adding the following to the module configruation:
+
+    <property name="severity" value="${optionalSeverity}"/>
+  -->
+  <property name="severity" value="error"/>
+
+  <!-- Never use tabs! -->
+  <module name="FileTabCharacter"/>
+
+  <!-- Always use UTF-8. -->
+  <property name="charset" value="UTF-8"/>
+
+  <module name="TreeWalker">
+    <!-- Always add the current ASF license header in all versioned files. -->
+    <module name="Regexp">
+      <property name="format" value="\A/\*\*\n \* Licensed to Apereo under one or more contributor license\n \* agreements\. See the NOTICE file distributed with this work\n \* for additional information regarding copyright ownership\.\n \* Apereo licenses this file to you under the Apache License,\n \* Version 2\.0 \(the &quot;License&quot;\); you may not use this file\n \* except in compliance with the License\.  You may obtain a\n \* copy of the License at the following location:\n \*\n \*   http://www\.apache\.org/licenses/LICENSE-2\.0\n \*\n \* Unless required by applicable law or agreed to in writing,\n \* software distributed under the License is distributed on an\n \* &quot;AS IS&quot; BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY\n \* KIND, either express or implied\.  See the License for the\n \* specific language governing permissions and limitations\n \* under the License\.\n \*/"/>
+      <property name="message" value="ASF license header not found"/>
+    </module>
+
+    <!-- Remove all trailing whitespaces. -->
+    <module name="Regexp">
+      <property name="format" value="[ \t]+$"/>
+      <property name="illegalPattern" value="true"/>
+      <property name="message" value="Trailing whitespace"/>
+    </module>
+
+    <!-- Always use 4 space indents and never use tabs! -->
+    <module name="Indentation"/>

```

Comment:
Someday we'll get to [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html) and it'll be [two space indents](https://google.github.io/styleguide/javaguide.html#s4.2-block-indentation). :smile: 

What with this being a new project, or at least a new module with a distinct codebase, this could be that day.